app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partnership/PartnershipOtherPartnerContactNameController.scala

```diff
@@ -54,80 +54,80 @@ class PartnershipOtherPartnerContactNameController @Inject() (
 
   def displayPage(): Action[AnyContent] =
     (authenticate andThen journeyAction) { implicit request =>
+      // TODO will have been populated by the preceding GRS journey
       val inflightOtherPartner =
         Some(
           Partner(organisationName = Some("Partner organisation"),
                   partnerType = Some(PartnerTypeEnum.SOLE_TRADER)
-          ) // TODO will have been populated by the preceding GRS journey
+          )
         )
 
-      (for {
-        inflight         <- inflightOtherPartner
-        organisationName <- inflight.organisationName
-      } yield (for {
-        contactDetails <- inflight.contactDetails
-        firstName      <- contactDetails.firstName
-        lastName       <- contactDetails.lastName
+      inflightOtherPartner.map { inflight =>
+        inflight.organisationName.map { organisationName =>
+          val existingNameFields = for {
+            contactDetails <- inflight.contactDetails
+            firstName      <- contactDetails.firstName
+            lastName       <- contactDetails.lastName
+          } yield (firstName, lastName)
```

Comment:
I thought we were ditching these in preference of `map()` and `flatMap()` for the sake of consistency? I can certainly live with the `for yield` construct if you feel strongly that it improves readability here?

Reply:
This specific one I think is a good use of a for comprehension; turning a bunch of chained optionals into something which needs more than a single value.